lib/msal-browser/src/interaction_client/StandardInteractionClient.ts

```diff
@@ -277,7 +277,11 @@ export abstract class StandardInteractionClient extends BaseInteractionClient {
             if (adalIdTokenString) {
                 const adalIdToken = new IdToken(adalIdTokenString, this.browserCrypto);
                 this.browserStorage.removeItem(PersistentCacheKeys.ADAL_ID_TOKEN);
-                if (adalIdToken.claims && adalIdToken.claims.upn) {
+                if (adalIdToken.claims && adalIdToken.claims.preferred_username) {
+                    this.logger.verbose("No SSO params used and ADAL token retrieved, setting ADAL preferred_username as loginHint");
+                    validatedRequest.loginHint = adalIdToken.claims.preferred_username;
+                }
+                else if (adalIdToken.claims && adalIdToken.claims.upn) {
                     this.logger.verbose("No SSO params used and ADAL token retrieved, setting ADAL upn as loginHint");
                     validatedRequest.loginHint = adalIdToken.claims.upn;
                 }
```

Comment:
Seeing this now, I think it would be a useful to add an `else` block here, and print a message indicating that an ADAL token was found, but it did not have proper claims to set a login Hint.

Something like:

```suggestion
                } else {
                    this.logger.verbose("No SSO params used and ADAL token retrieved, however, no account hint claim found. Enable preferred_username or upn id token claim to get SSO.");
                }
```